lib/Plack/Response.pm

```diff
@@ -123,47 +124,25 @@ sub _body {
 sub _finalize_cookies {
     my($self, $headers) = @_;
 
-    while (my($name, $val) = each %{$self->cookies}) {
-        my $cookie = $self->_bake_cookie($name, $val);
-        push @$headers, 'Set-Cookie' => $cookie;
-    }
-}
-
-sub _bake_cookie {
-    my($self, $name, $val) = @_;
-
-    return '' unless defined $val;
-    $val = { value => $val } unless ref $val eq 'HASH';
-
-    my @cookie = ( URI::Escape::uri_escape($name) . "=" . URI::Escape::uri_escape($val->{value}) );
-    push @cookie, "domain=" . $val->{domain}   if $val->{domain};
-    push @cookie, "path=" . $val->{path}       if $val->{path};
-    push @cookie, "expires=" . $self->_date($val->{expires}) if $val->{expires};
-    push @cookie, "max-age=" . $val->{"max-age"} if $val->{"max-age"};
-    push @cookie, "secure"                     if $val->{secure};
-    push @cookie, "HttpOnly"                   if $val->{httponly};
+    foreach my $name ( keys %{ $self->cookies } ) {
+        my $val = $self->cookies->{$name};
 
-    return join "; ", @cookie;
-}
-
-my @MON  = qw( Jan Feb Mar Apr May Jun Jul Aug Sep Oct Nov Dec );
-my @WDAY = qw( Sun Mon Tue Wed Thu Fri Sat );
-
-sub _date {
-    my($self, $expires) = @_;
+        my @cookie_args = (
+            'domain',
+            'expires',
+            'httponly',
+            'max-age',
+            'path',
+            'secure',
+            'value',
+        );
 
-    if ($expires =~ /^\d+$/) {
-        # all numbers -> epoch date
-        # (cookies use '-' as date separator, HTTP uses ' ')
-        my($sec, $min, $hour, $mday, $mon, $year, $wday) = gmtime($expires);
-        $year += 1900;
-
-        return sprintf("%s, %02d-%s-%04d %02d:%02d:%02d GMT",
-                       $WDAY[$wday], $mday, $MON[$mon], $year, $hour, $min, $sec);
+        my %cookie
+            = map { $_ => $val->{$_} } grep { $val->{$_} } @cookie_args;
```

Comment:
This copy is not required.

`Plack::Response->cookies` allows scalar values.
see https://metacpan.org/pod/Plack::Response#cookies

```
$res->cookies->{foo} = 123;
$res->cookies->{foo} = { value => '123' };
```